File: javascript/difference-of-squares/difference_of_squares.js
function Squares(size){
  var numbers =  Array.apply(null, {length: size+1}).map(Number.call, Number);
  this.squareOfSums = square(sum(numbers));
  this.sumOfSquares = sum(numbers.map(square));
  this.difference = this.squareOfSums - this.sumOfSquares;
}

function sum(array){
  return array.reduce(function(a,b) { return a+b; });
}

function square(number){
  return Math.pow(number,2);
}

module.exports = Squares;

File: go/difference-of-squares/difference_of_squares.go
package diffsquares

const testVersion = 1

// SquareOfSums calculates the sum from 1 up to a given number and returns the square of that sum
func SquareOfSums(max int) int {
	return max * max * (max + 1) * (max + 1) / 4
}

// SumOfSquares sums the squares of numbers from one up to a given number
func SumOfSquares(max int) int {
	return max * (max + 1) * (2*max + 1) / 6
}

// Difference calculates the difference of the SquareOfSums and SumOfSquares
func Difference(max int) int {
	return SquareOfSums(max) - SumOfSquares(max)
}
